lib/std/private/dbutils.nim

```diff
@@ -0,0 +1,15 @@
+import db_common
+
+
+template dbFormatImpl*(formatstr: SqlQuery, dbQuote: proc (s: string): string, args: varargs[string]): string =
+  var res = ""
+  var a = 0
+  for c in items(string(formatstr)):
+    if c == '?':
+      if a == args.len:
+        dbError("""The number of parameter doesn't match that of "?" """)
```

Comment:
show the minimum sufficient runtime context in error msgs; eg:
```nim
dbError("The number of parameter doesn't match that of `?` " % [...])
```

Reply:
```suggestion
        dbError("""The number of query arguments given exceeds the number of parameters present in the query. """)
```
An even better error message would be to show how many arguments and how many parameters are present.

Reply:
Positive.

> dbError("""The number of query arguments given exceeds the number of parameters present in the query. """)

It should be good enough for this super corner case.